virtualfish/environment.fish

```diff
@@ -0,0 +1,56 @@
+function __vfext_environment_activate --on-event virtualenv_did_activate
+    if test -f $VIRTUAL_ENV/virtualfish-environment
+        for line in (cat $VIRTUAL_ENV/virtualfish-environment)
+            # Skip empty lines and comments
+            if not string length -q $line
+                or test (string sub -s 1 -l 1 $line) = "#"
+                continue
+            end
+
+            set key (echo $line | cut -d = -f 1)
+            set value (echo $line | cut -d = -f 2-)
+
+            # Preserve existing env var with shared name
+            if set -q $key
+                set -xg __VF_ENVIRONMENT_OLD_VALUE_$key $$key
+            end
+
+            # Eval to allow for expanding variables, e.g. PATH=$PATH foo
+            set -gx $key (eval echo $value)
+        end
+    end
+end
+
+function __vfext_environment_deactivate --on-event virtualenv_will_deactivate
+    if test -f $VIRTUAL_ENV/virtualfish-environment
+        for line in (cat $VIRTUAL_ENV/virtualfish-environment)
```

Comment:
Same comment here with regards to perhaps using Fish's `read` built-in command. Also, as above, it might be better to use `test -r` here, which ensures the file exists _and_ is readable.